Classes/Issues/Comments/Markdown/CommentModelsFromMarkdown.swift

```diff
@@ -404,20 +431,56 @@ func shortenGitHubLinks(attributedString: NSAttributedString,
     return mutableAttributedString
 }
 
+private let checkedIdentifier = ">/CHECKED>/"
+private let uncheckedIdentifier = ">/UNCHECKED>/"
+private let checkmarkRegex = try! NSRegularExpression(pattern: "^- (\\[([ |x])\\])", options: [.anchorsMatchLines])
+private func annotateCheckmarks(markdown: String) -> String {
+    let matches = checkmarkRegex.matches(in: markdown, options: [], range: markdown.nsrange)
+    let result = NSMutableString(string: markdown)
+    for match in matches.reversed() {
```

Comment:
`match.range(at:)` crashes if it doesn't have the range, for a bit of added security should we add a where clause to the for loop in order to prevent this?

`for match in matches.reversed() where match.numberOfRanges == 2 {`

Reply:
Tho by definition of the regex, nothing can match w/out 2 groups (nested)

Reply:
Fair point, really not a fan of the regex API in Swift tbh! Not very safe and often have to cast to NS[String] 😅

Reply:
Ya definitely showing its age... all the `NSRange` too 🙄